pkg/liqo-controller-manager/authentication/utils/nonce.go

```diff
@@ -55,7 +55,9 @@ func EnsureSignedNonceSecret(ctx context.Context, cl client.Client,
 			return fmt.Errorf("nonce not provided and no nonce secret found")
 		}
 		secret := forge.SignedNonce(remoteClusterID, tenantNamespace, *nonce)
-		if err := cl.Create(ctx, secret); err != nil {
+		if _, err := resource.CreateOrUpdate(ctx, cl, secret, func() error {
+			return nil
+		}); err != nil {
 			return fmt.Errorf("unable to create nonce secret: %w", err)
 		}
 		return nil
```

Comment:
I find this potentially risky or misused. If the object already exists, the CreateOrUpdate does not apply the previously forged object (since it only gets the exisiting object and the passed mutateFn does not do anything). I wonder if it's safer to use a proper Create() wrapper (or just add the global labels before). 
Note: of course, for this particular case it works, as we check before that the secret does not exist